### PR TITLE
[kuberay] Update KubeRay operator commit, turn autoscaler RPC drain back on

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Related issue number

This PR:

- Updates the KubeRay operator commit used in the Ray CI autoscaling test
- Uses the RayCluster autoscaling sample config from the KubeRay repo in place of of a config from the Ray repo
- Turns the autoscaler RPC worker drain back on, as I saw some dead node messages from the GCS, and the RPC drain is supposed to avoid those. 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/autoscaler/_private/kuberay/autoscaling_config.py [hunk 1]
```diff
@@ -349,12 +349,10 @@ def _get_custom_resources(
 ) -> Dict[str, int]:
     """Format custom resources based on the `resources` Ray start param.
 
-    For the current prototype, the value of the `resources` field must
+    Currently, the value of the `resources` field must
     be formatted as follows:
     '"{\"Custom1\": 1, \"Custom2\": 5}"'.
 
-    We intend to provide a better interface soon.
```
Comment (DmitriGekhtman) [inline python/ray/autoscaler/_private/kuberay/autoscaling_config.py:356]: We decided to leave it as is for now.

File: python/ray/autoscaler/_private/kuberay/autoscaling_config.py [hunk 2]
```diff
@@ -157,7 +157,7 @@ def _generate_provider_config(ray_cluster_namespace: str) -> Dict[str, Any]:
         DISABLE_LAUNCH_CONFIG_CHECK_KEY: True,
         FOREGROUND_NODE_LAUNCH_KEY: True,
         WORKER_LIVENESS_CHECK_KEY: False,
-        WORKER_RPC_DRAIN_KEY: False,
+        WORKER_RPC_DRAIN_KEY: True,
```
Comment (wuisawesome) [inline python/ray/autoscaler/_private/kuberay/autoscaling_config.py:160]: Comment explaining the container restart race condition pls

**Review by wuisawesome — APPROVED** [review]

File: python/ray/autoscaler/_private/kuberay/autoscaling_config.py (hunk 2 above)
Comment (DmitriGekhtman) [reply python/ray/autoscaler/_private/kuberay/autoscaling_config.py:160]: ok

Comment (DmitriGekhtman) [reply python/ray/autoscaler/_private/kuberay/autoscaling_config.py:168]: Added a comment.

**Comment by DmitriGekhtman** [comment]
Operator test passes, merging.

**Pull request merged**